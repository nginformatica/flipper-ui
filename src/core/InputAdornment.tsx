import { InputAdornment as MuiInputAdornment } from '@material-ui/core'
import React, { FC } from 'react'
import { DefaultProps } from './types'

export interface InputAdornmentProps extends DefaultProps {
    position: 'start' | 'end'
}

const InputAdornment: FC<InputAdornmentProps> = ({
    children,
    ...otherProps
}) => <MuiInputAdornment {...otherProps}>{children}</MuiInputAdornment>

export default InputAdornment
