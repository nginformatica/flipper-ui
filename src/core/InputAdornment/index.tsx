import { InputAdornment as MuiInputAdornment } from '@material-ui/core'
import React from 'react'
import { DefaultProps } from '../types'

export interface InputAdornmentProps extends DefaultProps {
    position: 'start' | 'end'
}

const InputAdornment = ({ children, ...otherProps }: InputAdornmentProps) => (
    <MuiInputAdornment {...otherProps}>{children}</MuiInputAdornment>
)

export default InputAdornment
