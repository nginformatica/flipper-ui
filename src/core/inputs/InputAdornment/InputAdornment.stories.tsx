import React from 'react'
import { Meta, StoryFn } from '@storybook/react'
import InputAdornment from '.'

export default {
    title: 'Inputs/InputAdornment',
    component: InputAdornment
} as Meta<typeof InputAdornment>

const Template: StoryFn<typeof InputAdornment> = args => (
    <InputAdornment {...args} />
)
export const Default = Template.bind({})

Default.args = {
    children: 'InputAdornment'
}
