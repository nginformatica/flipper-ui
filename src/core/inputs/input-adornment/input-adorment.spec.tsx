import React from 'react'
import { render, screen } from '@testing-library/react'
import InputAdornment from '.'

describe('InputAdornment', () => {
    it('should render on start position', () => {
        render(
            <InputAdornment
                data-testid='input-adornment-container'
                position='start'>
                <>InputAdornment</>
            </InputAdornment>
        )

        const inputAdornment = screen.getByText('InputAdornment')
        const container = screen.getByTestId('input-adornment-container')

        expect(inputAdornment).toBeDefined()
        expect(container).toBeDefined()
        expect(inputAdornment.classList).toContain(
            'MuiInputAdornment-positionStart'
        )
    })

    it('should render on end position', () => {
        render(
            <InputAdornment
                data-testid='input-adornment-container'
                position='end'>
                <>InputAdornment</>
            </InputAdornment>
        )

        const inputAdornment = screen.getByText('InputAdornment')
        const container = screen.getByTestId('input-adornment-container')

        expect(inputAdornment).toBeDefined()
        expect(container).toBeDefined()
        expect(inputAdornment.classList).toContain(
            'MuiInputAdornment-positionEnd'
        )
    })

    it('should match snapshot', () => {
        const { container: start } = render(
            <InputAdornment position='start'>
                <>InputAdornment</>
            </InputAdornment>
        )

        const { container: end } = render(
            <InputAdornment position='end'>
                <>InputAdornment</>
            </InputAdornment>
        )

        expect(start).toMatchSnapshot()
        expect(end).toMatchSnapshot()
    })
})
