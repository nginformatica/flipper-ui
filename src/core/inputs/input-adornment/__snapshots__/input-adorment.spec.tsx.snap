// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAdornment should match snapshot 1`] = `
<div>
  <div
    class="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
  >
    <span
      class="notranslate"
    >
      â€‹
    </span>
    InputAdornment
  </div>
</div>
`;

exports[`InputAdornment should match snapshot 2`] = `
<div>
  <div
    class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
  >
    InputAdornment
  </div>
</div>
`;
